src/YiiSentryConfig.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Sentry;
+
+use InvalidArgumentException;
+
+final class YiiSentryConfig
+{
+    public function __construct(protected array $config)
+    {
+    }
+
+    /**
+     * Check if a DSN was set in the config.
+     */
+    public function hasDsnSet(): bool
+    {
+        $config = $this->getUserConfig();
+
+        return !empty($config['options']['dsn']);
+    }
+
+    /**
+     * Retrieve the user configuration.
+     */
+    public function getUserConfig(): array
+    {
+        $config = $this->config;
+
+        return empty($config) ? [] : $config;
+    }
+
+    public function getOptions(): array
+    {
+        if (empty($this->config['options'])) {
+            return [];
+        }
+        return is_array($this->config['options'])
```

Comment:
Could be reused with different key names.

Reply:
do not understand

Reply:
I meant that this part is repeated multiple times. Could be extracted to a separate function with key as parameter for example.